spark/src/main/scala/ai/chronon/spark/Extensions.scala

```diff
@@ -154,10 +156,31 @@ object Extensions {
       TableUtils(df.sparkSession).insertUnPartitioned(df, tableName, tableProperties)
     }
 
-    def prefixColumnNames(prefix: String, columns: Seq[String]): DataFrame = {
-      columns.foldLeft(df) { (renamedDf, key) =>
-        renamedDf.withColumnRenamed(key, s"${prefix}_$key")
+    def padFields(structType: sql.types.StructType): DataFrame = {
+      val existingColumns = df.columns.toSet
+      val paddedColumns = structType
+        .filterNot(field => existingColumns.contains(field.name))
+        .map(field => lit(null).cast(field.dataType).as(field.name))
```

Comment:
will this force non-nullable fields to be nullable?

Reply:
Yes, but this is not changing existing behavior. That's how the current implementation works. I added a commit that removes the current implementation of this method (I missed it initially), you can see from the diff that [the logic is the same](https://github.com/airbnb/chronon/pull/888/files#diff-b5033868ee521b9f585222ce5c40939392df053bbca36437c1d0757f67864f9dL80).

I agree this would be more correct if it respects the "nullable" definition in the schema, probably padding with a type-specific default value instead in those cases. But that's a more involved change, so I would prefer to leave it out of scope. The purpose of this PR is just to address the performance issue, not to change existing behavior.

Reply:
ah ok thank you for including that commit!